src/index.ts

```diff
@@ -1,4 +1,137 @@
-export default () => {
-  // eslint-disable-next-line no-console
-  console.log('https://www.youtube.com/watch?v=dQw4w9WgXcQ')
+import ResponseError from './ResponseError'
+
+export interface ApiClientConfiguration {
+  baseUrl: string
+  headers: Record<string, string>
 }
+
+export interface RequestOptions {
+  baseUrl?: string
+  headers?: Record<string, string>
+  params?: Record<string, string | number>
+  accessToken?: string
+}
+
+class ApiClient {
+  private configuration: ApiClientConfiguration = {
+    baseUrl: '',
+    headers: { 'Content-Type': 'application/json' },
+  }
+
+  private static replaceParameters(path: string, options?: RequestOptions) {
```

Comment:
Same as i commented on another one I would always use an object instead of multiple params